waku/protocol/v2/waku_filter.nim

```diff
@@ -15,14 +15,22 @@ import
 # relay protocol.
 
 const
-  WakuFilterCodec* = "/vac/waku/filter/2.0.0-alpha2"
+  WakuFilterCodec* = "/vac/waku/filter/2.0.0-alpha3"
 
 type
   ContentFilter* = object
     topics*: seq[string]
 
+  FilterRequest* = object
+    contentFilter*: seq[ContentFilter] 
+    topic*: string
+
+  MessagePush* = object
+    message*: seq[Message]
+
   FilterRPC* = object
```

Comment:
I guess I missed this in specs review and earlier PR reviews on storage protocol also, but why do we need this object?
And further more, why do we need the encode and init of `FilterRpc`?
Perhaps I'm misunderstanding the working of libp2p, but wouldn't the actual RPC messages (FilterRequest and MessagePush) be enough? We don't need to put this in "one" RPC message...?

Perhaps there is some confusion with the RPC message in GossipSub that is called `RPC` and which contains more than just one type of data (subscriptions and the actual messages being published). I believe that is done so subscription data can be tagged along with messages being pushed. But I don't think that is something we need to do here in a 1 to 1 protocol.

Reply:
Oh you are right @kdeme, that's my bad. I think this came from me moving over RPCs from previous protocol, and then I forgot about it. I guess an argument could be made that it allows us to reuse things like request id, but it might make sense to KISS and have it at the flat level

Reply:
You could still decide to reuse the `Message` structure on top, but I guess something more basic is enough (req id will be useful, anything regarding `from` and `signature` probably not?). My point was however mostly about having 2 actual separate RPC calls.

Reply:
@decanus since you are working on this code, do you have a preference for how we structure it? should be a minimal spec change. IMO either:

1. Flatten messages out, or
2. Pull out RPC protobuf and re-use it for filter and store

Reply:
@oskarth I have no real preference, however for not redoing work consistently I'd like to leave it as is for now. Especially with the fact that this PR is quite a large refactor of the message format.

Reply:
Fair, perhaps we can pull it out in an enhancement issue?

Reply:
I am fine with that. I don't fully understand the problem however.

Reply:
Which part doesn't make sense?